Mono.Debugging/Mono.Debugging.Client/BreakpointStore.cs

```diff
@@ -476,7 +476,15 @@ static string ResolveFullPath (string path)
 			try {
 				buffer = Marshal.AllocHGlobal (PATHMAX);
 				var result = realpath (path, buffer);
-				return result == IntPtr.Zero ? "" : Marshal.PtrToStringAuto (buffer);
+				var realPath = result == IntPtr.Zero ? "" : Marshal.PtrToStringAuto (buffer);
+
```

Comment:
I wonder if we could clean this up a bit by doing:

```csharp
if (result == IntPtr.Zero && !File.Exists (path))
```

Reply:
I think that depends on whether realpath can return Intptr.Zero and the file does exist. In that case we'd try to marshal a string from a zero pointer. the current code handles that case presently (returning an empty string).